script_lib.py

```diff
@@ -0,0 +1,347 @@
+        True
+        '''
+        return sorted(set(s.sql for s in self.dep_closure()))
+
+    @classmethod
+    def _get_deps(cls, sql: Text) -> List['SQLMixin']:
+        '''
+        >>> ds = Script._get_deps(
+        ...     "select col from t where 'dep' = 'oops.sql'")
+        Traceback (most recent call last):
+            ...
+        KeyError: 'oops'
+
+        >>> Script._get_deps(
+        ...     "select col from t where x = 'name.sql'")
+        []
+        '''
+        from typing import cast
+
+        m = re.search(r"select \S+ from \S+ where 'dep' = '([^']+)'", sql)
+        if not m:
+            return []
+        name, ext = m.group(1).rsplit('.', 1)
+        choices = Script if ext == 'sql' else []
+        deps = [cast(SQLMixin, s) for s in choices if s.name == name]
+        if not deps:
+            raise KeyError(name)
+        return deps
+
+    @classmethod
+    def sqlerror(cls, s: SQL) -> Optional[bool]:
+        if s.strip().lower() == 'whenever sqlerror exit':
+            return False
+        elif s.strip().lower() == 'whenever sqlerror continue':
+            return True
+        return None
+
+
+class ScriptMixin(SQLMixin):
+    @property
+    def extension(self) -> str:
+        return '.sql'
+
+    def parse(self, text: SQL,
+              block_sep: str='/') -> Iterable[StatementInContext]:
+        lines = [l.strip() for l in text.split('\n')]
+        return (sql_syntax.iter_blocks(text) if block_sep in lines
+                else iter_statement(text))
+
+    def created_objects(self) -> List[ObjectId]:
+        return [obj
+                for _l, _comment, stmt in iter_statement(self.sql)
+                for obj in sql_syntax.created_objects(stmt)]
+
+    def inserted_tables(self,
+                        variables: Optional[Environment]={}) -> List[Name]:
+        return [obj
+                for _l, _comment, stmt in iter_statement(self.sql)
+                for obj in sql_syntax.inserted_tables(
+                        sql_syntax.substitute(stmt, self._all_vars(variables)))]
+
+
+class Script(ScriptMixin, enum.Enum):
+    '''Script is an enum.Enum of contents.
+
+    ISSUE: It's tempting to consider separate libraries for NAACCR,
+           NTDS, etc., but that doesn't integrate well with the
+           generic luigi.EnumParameter() in etl_tasks.SqlScriptTask.
+
+    '''
+    [
+        # Keep sorted
+        condition,
+        death,
+        death_cause,
+        demographic,
+        diagnosis,
+        dispensing,
+        encounter,
+        enrollment,
+        harvest,
+        lab_result_cm,
+        med_admin,
+        med_admin_init,
+        obs_clin,
+        obs_gen,
+        pcornet_init,
+        pcornet_loader,
+        pcornet_trail,
+        prescribing,
+        pro_cm,
+        procedures,
+        provider,
+        vital
+    ] = [
+        pkg.resource_string(__name__,
+                            'Oracle/' + fname).decode('utf-8')
+        for fname in [
```

Comment:
The bulk of the Luigi code is borrowed from Grouse.  The changes here include all of the new SQL scripts needed to build the full CDM table set.  As mentioned in i2p_tasks.py, only med_admin is needed in the initial deliverable.

Reply:
I see why you were more inclined to add such comments here... this is more about status than design.